servicetalk-grpc-protoc/src/main/java/io/servicetalk/grpc/protoc/Generator.java

```diff
@@ -250,26 +222,50 @@ private void addSerializationProviderInit(final State state, final TypeSpec.Buil
 
     private void addServiceRpcInterfaces(final State state, final TypeSpec.Builder serviceClassBuilder) {
         state.serviceRpcInterfaces = new ArrayList<>(2 * state.serviceProto.getMethodCount());
-
-        state.serviceProto.getMethodList().forEach(methodProto -> Stream.of(false, true).forEach(blocking -> {
-            final String name = context.deconflictJavaTypeName((blocking ? Blocking : "") +
+        state.serviceProto.getMethodList().forEach(methodProto -> {
+            final String name = context.deconflictJavaTypeName(
```

Comment:
What was the reason to decouple async and blocking interfaces generation? The code for them is almost similar. With your changes you will need to do different `pathSpecBuilder.initializer` depending on the `blocking` flag. For everything else all conditions were preexisting. Consider reverting the previous logic.

Reply:
The primary motivation is to have the ability to refer to the corresponding async RPC interface `PATH` constant. Since we deconflict the name, there is no way to get the same name as was used by the async RPC interface apart from what I am doing here.
I can extract common code between these two lambdas into a separate method if you prefer but we will have to do the following:
- Iterate over the methods in proto file to generate async interfaces.
- Iterate over the async interfaces to generate the corresponding blocking interfaces.

This also has a nice side-effect that async and blocking interfaces for the same RPC method has similar names. eg: if the RPC method name is `foo` but the service name is also `foo` then with existing code, async RPC name will be `Foo0` but blocking interface will be `BlockingFoo`. After this change the names will be `Foo0` and `BlockingFoo0` which is nicer.

Reply:
Agreed, thanks for clarification.